modules/Core/hooks/ContentHook.php

```diff
@@ -17,10 +17,10 @@ class ContentHook extends HookBase {
      * @param array $params
      * @return array
      */
-    public static function codeTransform(array $params = []): array {
-        if (parent::validateParams($params, ['content'])) {
+    public static function codeTransform(AbstractEvent $event): void {
+        if (isset($event->content)) {
```

Comment:
This won't work, AbstractEvent has no params set on it directly.

We perhaps should introduce a `AbstractContentEvent` which builtin events (such as your new `RenderPostEvent`) can extend to remove duplication of a `content `property, and make these content hooks more explicit.